conjure-http/src/server/runtime.rs

```diff
@@ -0,0 +1,391 @@
+// Copyright 2024 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//! Runtime configuration for Conjure servers.
+
+use crate::server::Encoding;
+use crate::server::JsonEncoding;
+use crate::server::SmileEncoding;
+use conjure_error::Error;
+use conjure_error::InvalidArgument;
+use http::header::ACCEPT;
+use http::header::CONTENT_TYPE;
+use http::HeaderMap;
+use mediatype::names;
+use mediatype::MediaType;
+use mediatype::MediaTypeList;
+use mediatype::ReadParams;
+
+/// A type providing server logic that is configured at runtime.
+pub struct ConjureRuntime {
+    encodings: Vec<Box<dyn Encoding + Sync + Send>>,
+}
+
+impl ConjureRuntime {
+    /// Creates a new runtime with default settings.
+    pub fn new() -> Self {
+        Self::builder().build()
```

Comment:
ok cool, in the builder adds json and smile if no encodings added explicitly

Reply:
Yeah the only behavior change unless you start messing with config is that the server will be willing to consume Smile requests and return Smile responses if asked, but it'll still default to JSON.